oauth2_provider/views/oidc.py

```diff
@@ -120,3 +147,246 @@ def _create_userinfo_response(self, request):
+    Validates the claims of an IDToken for use with OIDC RP-Initiated Logout.
+    """
+    validator = oauth2_settings.OAUTH2_VALIDATOR_CLASS()
+
+    # Verification of `iss` claim is mandated by OIDC RP-Initiated Logout specs.
+    if "iss" not in claims or claims["iss"] != validator.get_oidc_issuer_endpoint(request):
+        # IDToken was not issued by this OP, or it can not be verified.
+        return False
+
+    return True
+
+
+def validate_logout_request(request, id_token_hint, client_id, post_logout_redirect_uri):
+    """
+    Validate an OIDC RP-Initiated Logout Request.
+    `(prompt_logout, (post_logout_redirect_uri, application))` is returned.
+
+    `prompt_logout` indicates whether the logout has to be confirmed by the user. This happens if the
+    specifications force a confirmation, or it is enabled by `OIDC_RP_INITIATED_LOGOUT_ALWAYS_PROMPT`.
+    `post_logout_redirect_uri` is the validated URI where the User should be redirected to after the
+    logout. Can be None. None will redirect to "/" of this app. If it is set `application` will also
+    be set to the Application that is requesting the logout.
+
+    The `id_token_hint` will be validated if given. If both `client_id` and `id_token_hint` are given they
+    will be validated against each other.
+    """
+
+    id_token = None
+    must_prompt_logout = True
+    if id_token_hint:
+        # Only basic validation has been done on the IDToken at this point.
+        id_token, claims = _load_id_token(id_token_hint)
+
+        if not id_token or not _validate_claims(request, claims):
+            raise InvalidIDTokenError()
+
+        if id_token.user == request.user:
+            # A logout without user interaction (i.e. no prompt) is only allowed
+            # if an ID Token is provided that matches the current user.
+            must_prompt_logout = False
+
+        # If both id_token_hint and client_id are given it must be verified that they match.
+        if client_id:
+            if id_token.application.client_id != client_id:
+                raise ClientIdMissmatch()
+
+    # The standard states that a prompt should always be shown.
+    # This behaviour can be configured with OIDC_RP_INITIATED_LOGOUT_ALWAYS_PROMPT.
+    prompt_logout = must_prompt_logout or oauth2_settings.OIDC_RP_INITIATED_LOGOUT_ALWAYS_PROMPT
+
+    application = None
+    # Determine the application that is requesting the logout.
+    if client_id:
+        application = get_application_model().objects.get(client_id=client_id)
+    elif id_token:
+        application = id_token.application
+
+    # Validate `post_logout_redirect_uri`
+    if post_logout_redirect_uri:
+        if not application:
+            raise InvalidOIDCClientError()
+        scheme = urlparse(post_logout_redirect_uri)[0]
+        if not scheme:
+            raise InvalidOIDCRedirectURIError("A Scheme is required for the redirect URI.")
+        if scheme == "http" and application.client_type != "confidential":
```

Comment:
We might want to consider skipping this when running in DEBUG on localhost to facilitate local integration testing... @Qup42  what do you think?

Reply:
I understand the use case. But I couldn't find any other parts of the library where the logic is changed when running in `DEBUG`. The only thing I found is that more information is shown. I don't know if the convenience for local testing outweighs the reduced security if some deploys in `DEBUG` mode. Personally I would lean against doing that but I am also not completely opposed.

Reply:
I do extensive local development with docker hosting services. It is a very important use case for me. However I've been looking closely at https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout

```
post_logout_redirect_uri - OPTIONAL. URI to which the RP is requesting that the End-User's User Agent be redirected after a logout has been performed. This URI SHOULD use the https scheme and MAY contain port, path, and query parameter components; however, it MAY use the http scheme, provided that the Client Type is confidential, as defined in Section 2.1 of [OAuth 2.0](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RFC6749) [RFC6749], and provided the OP allows the use of http RP URIs. The URI MAY use an alternate scheme, such as one that is intended to identify a callback into a native application. The value MUST have been previously registered with the OP, either using the post_logout_redirect_uris Registration parameter or via another mechanism. An id_token_hint is also RECOMMENDED when this parameter is included.
```

technically 'SHOULD' is recommended with caveats... MUST is not used in this situation. So as long as one understands and has carefully weighed the impact it's okay to use HTTP instead of https. 


This is very similar to https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest

```
REQUIRED. Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider, with the matching performed as described in Section 6.2.1 of [[RFC3986]](https://openid.net/specs/openid-connect-core-1_0.html#RFC3986) (Simple String Comparison). When using this flow, the Redirection URI SHOULD use the https scheme; however, it MAY use the http scheme, provided that the Client Type is confidential, as defined in Section 2.1 of OAuth 2.0, and provided the OP allows the use of http Redirection URIs in this case. The Redirection URI MAY use an alternate scheme, such as one that is intended to identify a callback into a native application.
```

The scheme is not validated on the authorization request.  

What is validated on the request is the ALLOWED_REDIRECT_URI_SCHEMES and that the redirect_uri is present in the applications redirect_uris.  I feel like we could remove any explicit check for 'scheme' here and instead validate post_logout_redirect_uri on the application model and show a big warning note near the field saying that http schemes are strongly discouraged by the spec and pointing someone to the examples. 

For now I think we can just remove the https check.

Reply:
Your argument has convinced me enough that I would add this and make it active when in `DEBUG` mode.

Reply:
I have use cases where I use http://localhost:5147/ etc  as valid redirect URIs for doing local frontend development in react/svelte/vue etc against a production IDP and API. I would recommend we forego the https validation for post_logout_redirect_uri in the RPInitiatedLogoutView / validate_logout_request

Reply:
I've also confirmed that Auth0, Okta, and IdentityServer  allow http:// post_logout_redirect_uris, and based on code review solely, so does Keycloak.

Reply:
`http` is now allowed per default. Strict Checking can be enabled with `OIDC_RP_INITIATED_LOGOUT_STRICT_REDIRECT_URIS`.